packages/host/tests/acceptance/code-mode-test.ts

```diff
@@ -508,3 +595,15 @@ module('Acceptance | code mode tests', function (hooks) {
       .containsText('in Test Workspace B');
   });
 });
+
+async function elementIsVisible(element: Element) {
+  return new Promise((resolve) => {
+    let intersectionObserver = new IntersectionObserver(function (entries) {
```

Comment:
this looks really interesting, what is that?

Reply:
It’s usually used for detecting when elements come into view to produce infinite scroll or the like, but I employ it here just to see if something is visible without annoying nested element rectangle math 😆

Reply:
so cool!

Reply:
definitely, it can even detect when elements are _about_ to enter view (`rootMargin`) and what % of one is visible, it’s pretty nice!

Reply:
i wonder if we are using this in the boxel animation workspace--it seems super applicable...

Reply:
hmm it seems not, surprisingly!